src/AzureBlobStorageServiceProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+
+namespace A17\Twill;
+
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\ServiceProvider;
+use League\Flysystem\AzureBlobStorage\AzureBlobStorageAdapter;
+use League\Flysystem\Filesystem;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+
+class AzureBlobStorageServiceProvider extends ServiceProvider
+{
+    /**
+     * Perform post-registration booting of services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        Storage::extend('azure', function ($app, $config) {
+            $connectionString = sprintf(
+                'DefaultEndpointsProtocol=%s;AccountName=%s;AccountKey=%s;EndpointSuffix=%s',
+                isset($config['protocol']) ? $config['protocol'] : 'https',
+                $config['account']['name'],
+                $config['account']['key'],
+                $config['endpoint-suffix']
+            );
+            $client = BlobRestProxy::createBlobService($connectionString);
+            return new Filesystem(new AzureBlobStorageAdapter($client, $config['container']));
+        });
+    }
+
+    /**
+     * Register bindings in the container.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->mergeConfigFrom(__DIR__ . '/../config/disks.php', 'filesystems.disks');
```

Comment:
This already happens in Twill main service provider: https://github.com/area17/twill/blob/7ca1b4bfef36995bb9ec5d3fda20170c5808a1bc/src/TwillServiceProvider.php#L213

Any reason this would be needed here?

Reply:
Any valable reason here 😅